libs/wingsdk/src/target-sim/counter.inflight.ts

```diff
@@ -38,7 +38,7 @@ export class Counter implements ICounterClient, ISimulatorResourceInstance {
   public async cleanup(): Promise<void> {}
 
   public async save(): Promise<void> {
-    await fs.promises.writeFile(
+    fs.writeFileSync(
```

Comment:
Is there a way to also add some lower level protection against corrupted state files?

Reply:
Not sure 🤷‍♂️

It depends how much we want to design the API (so far I'm trying to avoid over-designing it). We could provide some set of formal APIs like `saveState(key: str, value: Json)` which all use safe operations under the hood, and also add some `validateState()` method that resources can implement which will be called before the state is read, etc.

My current thinking is to support resources like `cloud.Bucket`, simulator resources need to be able to read/write to their respective state directories whenever they want, and so the resource is responsible if it crashes or corrupts state somehow. But there might be alternatives I'm not thinking of

Reply:
High level I think first we should try what we can do to stop these cases of data corruption (and fix the issue in the console, assuming there's a bug there -- not addressed by this PR) and if that's not possible then we can double down on resiliency